de.dashup.model/src/main/java/de/dashup/model/service/DashupService.java

```diff
@@ -392,4 +393,206 @@ public void processLayoutModeChanges(LayoutModeStructureDTO layoutModeStructureD
             sectionIndex++;
         }
     }
+
+    // --- DRAFTS --- \\
+
+    public void getUsersDrafts(User user) throws SQLException {
+        Map<String, Object> whereParameters = new HashMap<>();
+        whereParameters.put("user_id", user.getId());
+        whereParameters.put("visibility", false);
+
+        List<Draft> drafts = new ArrayList<>();
+        for (DatabaseObject databaseObject : this.database.getObject(Database.Table.PANELS, DatabaseWidget.class, whereParameters)) {
+            drafts.add(new Draft().fromDatabaseObject(databaseObject));
+        }
+        user.setDrafts(drafts);
+    }
+
+    public Draft createDraft(User user, String draftName) throws SQLException {
+        Map<String, Object> values = new HashMap<>();
+        values.put("user_id", user.getId());
+        values.put("name", draftName);
+        values.put("visibility", false);
+
+        this.database.insert(Database.Table.PANELS, values);
+
+        Draft draft = new Draft();
+        draft.setId(this.database.getLatestId(Database.Table.PANELS));
+        draft.setName(draftName);
+
+        return draft;
+    }
+
+    public void updateWidgetInformation(Widget widget) throws SQLException, MissingInformationException, InvalidCodeException {
+        this.updateWidgetInformation(widget, true);
+    }
+
+    public void updateWidgetInformation(Widget widget, boolean checkAfterUpdate) throws SQLException, MissingInformationException, InvalidCodeException {
+        if (!checkAfterUpdate) {
+            checkWidget(widget);
+        }
+
+        Map<String, Object> whereParameters = new HashMap<>();
+        whereParameters.put("id", widget.getId());
+
+        Map<String, Object> values = new HashMap<>();
+        if (widget.getName() != null) {
+            values.put("name", widget.getName());
+        }
+        if (widget.getCodeSmall() != null) {
+            values.put("code_small", widget.getCodeSmall());
+        }
+        if (widget.getCodeMedium() != null) {
+            values.put("code_medium", widget.getCodeMedium());
+        }
+        if (widget.getCodeLarge() != null) {
+            values.put("code_large", widget.getCodeLarge());
+        }
+        if (widget.getShortDescription() != null) {
+            values.put("short_description", widget.getShortDescription());
+        }
+        if (widget.getDescription() != null) {
+            values.put("descriptions", widget.getDescription());
+        }
+        if (widget.getCategoryObject() != null) {
+            values.put("category", widget.getCategory());
+        }
+
+        this.database.update(Database.Table.PANELS, whereParameters, values);
+
+        if (widget.getTags().size() > 0) {
```

Comment:
You are using the widget tag count to test if an update is necessary, but if i delete all tags this count is 0 and you are updating nothing even if there are some tags that should be deleted.